src/scatter_buildblock/ScatterEstimation.cxx

```diff
@@ -0,0 +1,1486 @@
+
+        FilePath tmp(this->input_projdata_filename);
+        std::string out_filename = extras_path.get_path() + tmp.get_filename_no_extension() + "_2d.hs";
+
+        this->input_projdata_2d_sptr.reset(new ProjDataInterfile(this->input_projdata_sptr->get_exam_info_sptr(),
+                                                                 proj_data_info_2d_sptr,
+                                                                 out_filename,
+                                                                 std::ios::in | std::ios::out | std::ios::trunc));
+        SSRB(*this->input_projdata_2d_sptr,
+             *input_projdata_sptr,false);
+    }
+    else
+    {
+        input_projdata_2d_sptr = input_projdata_sptr;
+    }
+
+#else
+    {
+        std::string tmp_input2D = "./extras/nema_proj_f1g1d0b0.hs_2d.hs";
+        this->input_projdata_2d_sptr =
+                ProjData::read_from_file(tmp_input2D);
+    }
+#endif
+info("ScatterEstimation: Setting up reconstruction method ...");
+
+if(is_null_ptr(this->reconstruction_template_sptr))
+{
+    warning("ScatterEstimation: Reconstruction method has not been initialised. Aborting.");
+    return Succeeded::no;
+}
+
+    // We have to check which reconstruction method we are going to use ...
+    shared_ptr<AnalyticReconstruction> tmp_analytic =
+            dynamic_pointer_cast<AnalyticReconstruction >(this->reconstruction_template_sptr);
+    shared_ptr<IterativeReconstruction<DiscretisedDensity<3, float> >> tmp_iterative =
+            dynamic_pointer_cast<IterativeReconstruction<DiscretisedDensity<3, float> > >(reconstruction_template_sptr);
+
+    if (!is_null_ptr(tmp_analytic))
+    {
+        if(set_up_analytic() == Succeeded::no)
+        {
+            warning("ScatterEstimation: set_up_analytic reconstruction failed. Aborting.");
+            return Succeeded::no;
+        }
+
+        this->iterative_method = false;
+    }
+    else if (!is_null_ptr(tmp_iterative))
+    {
+        if(set_up_iterative(tmp_iterative) == Succeeded::no)
+        {
+            warning("ScatterEstimation: set_up_iterative reconstruction failed. Aborting.");
+            return Succeeded::no;
+        }
+
+        this->iterative_method = true;
+    }
+    else
+    {
+        warning("ScatterEstimation: Failure to detect a method of reconstruction. Aborting.");
+        return Succeeded::no;
+    }
+
+    if(iterative_method)
+        this->current_activity_image_sptr.reset(tmp_iterative->get_initial_data_ptr());
+
+    //    if ( run_debug_mode )
+    //    {
+    //        std::string out_filename = extras_path.get_path() + "inital_activity_image";
+    //        OutputFileFormat<DiscretisedDensity < 3, float > >::default_sptr()->
+    //                write_to_file(out_filename, *this->current_activity_image_sptr);
+    //    }
+
+    // Based on the activity image zoom the attenuation
+    if (!is_null_ptr(current_activity_image_sptr))
```

Comment:
not 100% sure why we do this. Did we have it originally? I'd probably cut it. If we need it, can in any case be replaced by
```C++
   shared_ptr<DiscDens> new_atten_image_sptr(current_activity_image_sptr-get_empty_copy());
   zoom_image(*new_atten_image_sptr, *atten_image_sptr,  ZoomOptions::preserve_values); // will need to add the casting
   atten_image_sptr=new_atten_image_sptr;
```

Reply:
I am not sure which is the comment here. 
We need to make the attenuation image to have the same voxel size as the activity image. 

If you have set an initial activity image you cannot do this in the ScatterSimulation.